src/main/java/io/appium/java_client/service/local/AppiumServiceBuilder.java

```diff
@@ -217,13 +218,25 @@ public AppiumServiceBuilder withArgument(ServerArgument argument, String value)
             case "-g":
                 withLogFile(new File(value));
                 break;
+            case "--base-path":
+                serverArguments.put(argName,sanitizeBasePath(value));
+                break;
             default:
                 serverArguments.put(argName, value);
                 break;
         }
         return this;
     }
 
+    private static String sanitizeBasePath(String basePath) {
+        basePath = checkNotNull(basePath).trim();
+        if (StringUtils.isBlank(basePath) || basePath.isEmpty()) {
```

Comment:
no need to check for blank value after trim operation
```suggestion
        basePath = checkNotNull(basePath).trim();
        if (basePath.isEmpty()) {
```

Reply:
Noted on this, taken care in the latest commit.